metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -591,31 +620,6 @@ def _process_project(
                 )
                 continue
 
-        if self.config.include_table_lineage:
-            if (
-                self.config.store_last_lineage_extraction_timestamp
-                and self.redundant_run_skip_handler.should_skip_this_run(
-                    cur_start_time_millis=datetime_to_ts_millis(self.config.start_time)
-                )
-            ):
-                # Skip this run
-                self.report.report_warning(
-                    "lineage-extraction",
-                    f"Skip this run as there was a run later than the current start time: {self.config.start_time}",
-                )
-                return
```

Comment:
Previously, we'd short circuit if we were collecting lineage info and skipped the run. This is really confusing logic, because (a) it skips usage and profiling and (b) it's called once per project -- does this value ever actually change?

(a) is not that crazy for usage, because we do the same exact check, but we don't do this for profiling, so I think this makes it so whether you profile or not can be affected by `self.config.include_table_lineage` which doesn't seem right

Reply:
You are right. thanks for fixing this. Also I believe - the usage and operations aspect generation stage should also follow the same pattern as lineage. We can tackle this in follow up PR.